tests/test_signing.py

```diff
@@ -78,30 +79,65 @@ def test_different_keys_are_not_equal(self, k2):
         k1 = SigningKey(b"\x00" * crypto_sign_SEEDBYTES)
         assert_not_equal(k1, k2)
 
-    @pytest.mark.parametrize("seed", [
+    @pytest.mark.parametrize("hseed", [
         b"77076d0a7318a57d3c16c17251b26645df4c2f87ebc0992ab177fba51db92c2a",
     ])
-    def test_initialization_with_seed(self, seed):
-        SigningKey(seed, encoder=HexEncoder)
+    def test_initialization_with_seed(self, hseed):
+        seed = binascii.unhexlify(hseed)
+        SigningKey(seed)
 
     @pytest.mark.parametrize(
-        ("seed", "_public_key", "message", "signature", "expected"),
+        ("hseed", "_public_key", "message", "signature", "expected"),
         ed25519_known_answers()
     )
-    def test_message_signing(self, seed, _public_key,
+    def test_message_signing(self, hseed, _public_key,
                              message, signature, expected):
-        signing_key = SigningKey(
-            seed,
-            encoder=HexEncoder,
-        )
+        seed = binascii.unhexlify(hseed)
+        signing_key = SigningKey(seed)
         signed = signing_key.sign(
             binascii.unhexlify(message),
-            encoder=HexEncoder,
         )
 
-        assert signed == expected
-        assert signed.message == message
-        assert signed.signature == signature
+        assert signed == binascii.unhexlify(expected)
+        assert signed.message == binascii.unhexlify(message)
+        assert signed.signature == binascii.unhexlify(signature)
+
+    @pytest.mark.parametrize("hseed", [
+        b"77076d0a7318a57d3c16c17251b26645df4c2f87ebc0992ab177fba51db92c2a",
+    ])
+    def test_deprecation_of_encoder_parameter(self, hseed):
+        with pytest.warns(PyNaclDeprecated):
+            SigningKey(hseed,
+                       encoder=HexEncoder,
+                       )
+        with pytest.warns(PyNaclDeprecated):
+            SigningKey(hseed,
+                       HexEncoder,
+                       )
+        sk = SigningKey.generate()
+        unsigned = b"A test message!"
+        with pytest.warns(PyNaclDeprecated):
+            sk.sign(unsigned,
+                    encoder=HexEncoder)
+        with pytest.warns(PyNaclDeprecated):
+            sk.sign(unsigned,
+                    HexEncoder)
+
+    @pytest.mark.parametrize("hseed", [
+        b"77076d0a7318a57d3c16c17251b26645df4c2f87ebc0992ab177fba51db92c2a",
+    ])
+    def test_raising_on_excess_encoder_parameter(self, hseed):
```

Comment:
These tests aren't really testing anything interesting with the new approach, correct?

Reply:
Right, they were redundant. Reverted the relevant commit.